### PR TITLE
[GLUTEN-2874][VL] support allowDecimalPrecisionLoss

### PR DESCRIPTION
**Pull request by zhouyuan**
## What changes were proposed in this pull request?

This patch allows user to enable or disable decimal precision loss in decimal arithmetic

~this patch depends on https://github.com/oap-project/velox/pull/394~

(Fixes: #2874)

## How was this patch tested?

existing UT

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 1]
```diff
@@ -471,14 +471,6 @@ object ExpressionConverter extends SQLConfHelper with Logging {
             replaceWithExpressionTransformerInternal(_, attributeSeq, expressionsMap)),
           expr)
       case b: BinaryArithmetic if DecimalArithmeticUtil.isDecimalArithmetic(b) =>
-        // PrecisionLoss=true: velox support / ch not support
-        // PrecisionLoss=false: velox not support / ch support
-        // TODO ch support PrecisionLoss=true
```
Comment (zhouyuan) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:476]: Note: should fallback on CK backend if allow precision loss = false

**Comment by kecookier** [comment]
@zhouyuan I have tested this PR in my situation, and it looks good to me.  Thanks!

**Comment by kecookier** [comment]
@zhouyuan Hi there, could you please let me know the current status of this PR? Will it be included in the release-1.2.0 version? Thanks!

**Review by marin-ma — APPROVED** [review]
Thanks!
